default/python/AI/ShipDesignAI.py

```diff
@@ -1552,6 +1558,12 @@ def _calculate_weapon_shots(self, weapon_part, ignore_species=False):
             species_modifier = 0
         return base + species_modifier + tech_bonus
 
+    def _calculate_fighter_launch_bay_rate(self, hangar_part):
+        hangar_name = hangar_part.name
+        base = AIDependencies.LAUNCH_BAY_BASE_CAPACITY
+        hangar_bonus = AIDependencies.HANGAR_LAUNCH_CAPACITY_MODIFIER_DICT.get(hangar_name, 0)
+        return base + hangar_bonus
```

Comment:
We want the ShipDesignAI to be as generic as possible and handle trivial content changes without modification. This function breaks if a scripter adds another bay part.

Reply:
Do you have an idea how to do that without adding specific launch bay part types?

Reply:
With something like 
```
def _calculate_fighter_launch_bay_rate(self, bay_part_name, hangar_name)
    base = get_part_type(bay_part_name).capacity
    hangar_bonus = AIDependencies.HANGAR_LAUNCH_CAPACITY_MODIFIER_DICT.get(hangar_name, 0)
    return base + hangar_bonus
```
You can get the ```hangar_name``` after all parts are processed from the set ```hangar_parts``` (but should verify that we actually have one and only one hangar part first).